app/controllers/admin_controller.rb

```diff
@@ -93,12 +97,38 @@ def show_instructor
       @role = Role.new(:id => nil, :name => '(none)')
     end
   end
+  
+  def show_super_admin
+    @user = User.find(params[:id])
+    if @user.role_id
+      @role = Role.find(@user.role_id)
+    else
+      @role = Role.new(:id => nil, :name => '(none)')
+    end
+  end
+  def show_admin
+    @user = User.find(params[:id])
+    if @user.role_id
+      @role = Role.find(@user.role_id)
+    else
+      @role = Role.new(:id => nil, :name => '(none)')
+    end
+  end
 
   def remove_instructor
     User.find(params[:id]).destroy
     redirect_to :action => 'list_instructors'
   end
 
+  def remove_administrator
```

Comment:
I've removed the remove super admins method. I put that method initially because it was given in our project description.

 The delete admin functionality was broken earlier. I tried to fix it by creating the remove_admin method. It is similar to the remove_instructor method.